pyproject.toml

```diff
@@ -56,7 +56,7 @@ crashtest = "^0.4.1"
 dulwich = "^0.20.46"
 filelock = "^3.8.0"
 html5lib = "^1.0"
-importlib-metadata = { version = "^4.4", python = "<3.10" }
+importlib-metadata = { version = "^5.1", python = "<3.10" }
```

Comment:
According to https://pypi.org/project/importlib-metadata/, 4.6 is included in python 3.10. If 4.6 is enough for Python 3.10, why should we require 5.1 for python < 3.10?

Reply:
no big deal either way, but IMO it's as well for application-type code to stay current. What is gained by allowing old versions of importlib-metadata?

Indeed we're currently explicitly excluding new versions of importlib-metadata, which seems a reasonable thing to happen by accident, but a strange thing to choose deliberately.

Reply:
Yeah, `^4.4` isn't good, either. Probably, it should be `>=4.4,<6`.

Re allowing old versions: If you install poetry in its own virtual env (as recommended) it doesn't matter. But then we could also pin dependencies. 🤷

Reply:
sure, it's a can of worms.  But for applications - and poetry really is intended to be installed in its own virtual environment - I don't see meaningful upside in allowing a broad range.  There certainly is downside: it would be some sort of commitment to support old versions of importlib-metadata (snd there is no testing to verify that this will work).

However I don't think this is important enough that I care to worry about it: if you're asking me to make a change, please say so.

Reply:
> There certainly is downside: it would be some sort of commitment to support old versions of importlib-metadata (snd there is no testing to verify that this will work).

Personally, I like to keep constraints as broad as possible and only restrict them if a version is known to be incompatible. (At least, experienced users can easily exclude a bad version by themselves, but cannot allow an excluded version as easy.)

It might be a while before we find the time to discuss our update strategy internally.  So yes, maybe just change it and I'll merge it.